src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,110 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stdout, stderr or any other stream.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @see http://www.php-fig.org/psr/psr-3/
+ */
+final class Logger extends AbstractLogger
+{
+    /**
+     * @var array
+     */
+    private static $defaultLevelToStream = array(
+        LogLevel::EMERGENCY => 'php://stderr',
+        LogLevel::ALERT => 'php://stderr',
+        LogLevel::CRITICAL => 'php://stderr',
+        LogLevel::ERROR => 'php://stderr',
+        LogLevel::WARNING => 'php://stdout',
+        LogLevel::NOTICE => 'php://stdout',
+        LogLevel::INFO => 'php://stdout',
+        LogLevel::DEBUG => false,
+    );
+
+    private $levelToStream;
+    private $formatPattern;
+
+    /**
+     * @var bool
+     */
+    private $errored = false;
+
+    /**
+     * @param array  $levelToStream
+     * @param string $formatPattern
+     */
+    public function __construct(array $levelToStream = array(), $formatPattern = '[%s] %s')
+    {
+        $this->levelToStream = $levelToStream + self::$defaultLevelToStream;
+        $this->formatPattern = $formatPattern;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log($level, $message, array $context = array())
+    {
+        if (!isset($this->levelToStream[$level])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $level));
+        }
+
+        if (!$this->errored) {
+            $this->errored = in_array($level, array(LogLevel::EMERGENCY, LogLevel::ALERT, LogLevel::CRITICAL, LogLevel::ERROR));
+        }
+
+        if ($this->levelToStream[$level]) {
+            file_put_contents($this->levelToStream[$level], sprintf($this->formatPattern, $level, $this->interpolate($message, $context)).PHP_EOL, FILE_APPEND);
+        }
+    }
+
+    /**
+     * Returns true when any messages have been logged at error levels.
+     *
+     * @return bool
+     */
+    public function hasErrored()
+    {
+        return $this->errored;
+    }
+
+    /**
+     * Interpolates context values into the message placeholders.
+     *
+     * @author PHP Framework Interoperability Group
+     *
+     * @param string $message
+     * @param array  $context
+     *
+     * @return string
+     */
+    private function interpolate($message, array $context)
+    {
+        // build a replacement array with braces around the context keys
+        $replace = array();
+        foreach ($context as $key => $val) {
+            if (!is_array($val) && (!is_object($val) || method_exists($val, '__toString'))) {
+                $replace[sprintf('{%s}', $key)] = $val;
+            }
+        }
+
+        // interpolate replacement values into the message and return
+        return strtr($message, $replace);
```

Comment:
what about this for perf?
```php
        $replace = array();
        foreach ($context as $key => $val) {
            if (!\is_array($val) && (!\is_object($val) || method_exists($val, '__toString'))) {
                $replace["{{$key}}"] = $val;
            }
        }

        // interpolate replacement values into the message and return
        return $replace ? strtr($message, $replace) : $message;
```

Reply:
Couldn't we reuse the same code as in [PsrLogMessageProcessor](https://github.com/Seldaek/monolog/blob/master/src/Monolog/Processor/PsrLogMessageProcessor.php) ?

Reply:
I've merged both proposals and added some perf optims. I'll open a PR to update existing implementations like the one in `ConsoleLogger`.